filters/fadein/fadein.go

```diff
@@ -236,12 +237,12 @@ func (p *postProcessor) Do(r []*routing.Route) []*routing.Route {
 			}
 
 			key := endpointKey(s, h)
-			detected := p.detected[key].when
+			detected := endpointregistry.GetDetectedTime(h)
 			if detected.IsZero() || endpointsCreated[key].After(detected) {
 				detected = now
 			}
 
-			ep.Detected = detected
+			endpointregistry.SetDetectedTime(h, detected)
```

Comment:
Looks like you decided to keep existing fadeIn machinery for now, didn't you?

We can also remove stale endpoints around  `delete(p.detected, key)`

Reply:
I think we need a dedicated post processor to track endpoints instead of plugging here

Reply:
Yeah, in the scope of the current PR I am going only to introduce endpointregistry. When this component will be ready, I will continue to work on #2381 as I will have a way to implement in a clean way.